tests/framework/results/test_interface.py

```diff
@@ -187,11 +205,13 @@ def test_register_stratified_observation(mocker):
     assert obs.results_updater is not None
     assert obs.results_formatter is not None
     assert obs.stratifications == stratifications
+    # Assert a StratifiedObservation so mypy knows .aggregator and .aggregator_sources exist
```

Comment:
I don't think this comment is strictly necessary. It's a reasonable thing to assert in a test?